beps/0006-scaffolder-action-rollback/README.md

```diff
@@ -31,36 +32,191 @@ Introducing the rollback to scaffolder actions provides the mean to come back to
 
 ## Motivation
 
-The aim is to help to mitigate the issue of manual clean up the partially created resources.
+Mitigate the issue of manual clean up from partially created resources during template execution failures. When a template execution fails after creating a number of resources but before completing, the system is left in an inconsistent state. Currently, there is no standardized way to clean up these partially created resources, leading to orphaned resources in external systems requiring manual cleanup.
 
 ### Goals
 
-- We will extend action's API to be able to run rollback for failed tasks
-- Rollback will be optional
+- Provide a standardized, optional rollback mechanism for scaffolder actions
+- Enable proper cleanup of resources created in third-party systems when template execution fails
+- Improve the developer experience by reducing manual intervention for cleanup after failures
+- Allow for targeted rollbacks of specific actions rather than only full template rollbacks
 
 ### Non-Goals
 
-We will not cover all built-in actions with a rollback functionality
+- Modifying existing actions to implement rollback (this proposal only provides the framework)
+- Adding automatic rollback triggers for all possible error scenarios
 
 ## Proposal
 
 Rollback is going to be performed:
 
-- when user manually decide to perform this action.
-- task has to be recovered and TaskRecoverStrategy set to 'rollback'
+- when a user manually decide to perform this action.
```

Comment:
Does this mean there can be a button on the task page to rollback a completed task even if it has finished?

Reply:
I'm thinking it would be a rollback option (button) presented to the end user only when a template failure occurs. We might not want to start by providing the end user the ability to do this though.